### ISSUE TITLE
Add support for ppc64le

### ISSUE THREAD

**Issue by helix-loop** [opensearch-project/OpenSearch#1303]
Building OpenSearch on ppc64le fails with the following error:

FAILURE: Build failed with an exception.

* Where:
Script 'xyz/OpenSearch-1.0.0/gradle/runtime-jdk-provision.gradle' line: 33

* What went wrong:
A problem occurred evaluating script.
> can not determine architecture from [ppc64le]

Java: openjdk-11-jdk

**Comment by CEHENKLE** [comment]
@bbarani flagging for infra....

**Comment by dblock** [comment]
This error comes from https://github.com/opensearch-project/OpenSearch/blob/0e9f74e35f1255cb9ec45be3d8960aad195a9f6e/buildSrc/src/main/java/org/opensearch/gradle/Architecture.java#L40 where we've never seen a ppc64le (the little-endian version of ppc64). 

@helix-loop Does returning X64 in that code result in a successful build/test/run?

**Comment by helix-loop** [comment]
@dblock It does. Run is successful, check is throwing 2 errors, but I am able to curl localhost:9200 with the following change:

```
package org.opensearch.gradle;

public enum Architecture {

    X64,
    ARM64;

    public static Architecture current() {
        final String architecture = System.getProperty("os.arch", "");
        switch (architecture) {
            case "amd64":
            case "x86_64":
            case "ppc64le":<----
                return X64;
            case "aarch64":
                return ARM64;
            default:
                throw new IllegalArgumentException("can not determine architecture from [" + architecture + "]");
        }
    }

}

```

**Comment by dblock** [comment]
@helix-loop What are the test failures?

I think you can PR this change along with some basic test(s).

**Comment by dbwiddis** [comment]
FYI, commentary on the above gradle check errors in #2341.  

TLDR: I believe they are Gradle and/or OpenJ9 bugs that may be fixed by upgrading version(s).

**Comment by dbwiddis** [comment]
Hi, @seth-priya, please do take this on!  I had assigned myself #2341 (ppc64le) some time ago but unassigned a few weeks ago when it became clear I didn't have time to do it -- and also the solution in #4001 is clearly a more comprehensive one.

It's all yours!

**Comment by dbwiddis** [comment]
Closed in #5459
